### DISCUSSION TITLE
How to loop to variable in test case?

### DISCUSSION THREAD

**Discussion by DungeonMaestro215** [hneemann/Digital#927] [Q&A] [answered]
Hello,

The following code snippet is throwing the error 'Variable iters not found.' Is there a way to make the bound of the loop a variable name instead of a hardcoded number?

CLK A B Y
0     0 0 0
let iters = 16;
bits(4, iters)  # No problem accessing iters here
loop(i, iters)  # Variable iters not found
	bits(4, i)
end loop

Let me know if there is any more information that you would like. I could also attach an example file.

**Comment by hneemann** [comment, marked as answer]
No, the loop range needs to be a constant.
